### ISSUE TITLE
Verifying C++ code fails due to missing LLVM dependencies

### ISSUE THREAD

**Issue by btj** [verifast/verifast#246]
The VeriFast nightly release for macOS fails with
```
Fatal error: exception (Failure
   "Cxx AST exporter error:\
  \ndyld: Library not loaded: /usr/local/opt/llvm/lib/libLLVM.dylib\
  \n  Referenced from: /Users/bartj/verifast/verifast-21.04-20-g797e8fcf/bin/vf-cxx-ast-exporter\
  \n  Reason: image not found\
  \n")
```
Please test in a clean environment (e.g. run `sudo mv /usr/local/Cellar /usr/local/_Cellar` and similar for other dirs, or run in a fresh VM).

The release for Windows also fails, with
```
VeriFast internal error:
(Failure
  "the Cxx frontend received a serialized result which was undefined.")
```

**Comment by NielsMommen** [comment]
It looks like the LLVM package installed from Homebrew, is built with ` LLVM_BUILD_LLVM_DYLIB` and `LLVM_LINK_LLVM_DYLIB` turned on. `libLLVM.dylib` contains all of LLVM's components in one shared library. This results in CMake linking the shared library instead of the components static libraries.

On Windows, we explicitly pass all components' paths to `target_link_libraries`. This is also possible on Unix and avoids the dynamic linking. The downside is that we have to manually search for add LLVM/Clang dependencies to our target: `zlib` and `ncurses`. The question is if these packages are available on every Unix system. This breaks the Linux build because some dependencies between the components are cyclic. We could resolve this by passing the components multiple times and in a specific order to `target_link_libraries`.

Another option is to use the LLVM/Clang release from GitHub instead of Homebrew. These releases seem to be built with both ` LLVM_BUILD_LLVM_DYLIB` and `LLVM_LINK_LLVM_DYLIB` turned off. The big advantage is that CMake will resolve all cyclic and external dependencies. However, for Ubuntu, no release is available for LLVM 13.0 on 18.04 (there is one for 16.04 and 20.04).

**Comment by btj** [comment]
Another option would be to simply ship the LLVM DLL along with VeriFast, like we do with GTK and Z3? The launch script sets `DYLD_LIBRARY_PATH` to point to VeriFast's `lib` directory.

Of course, ideally we avoid shipping duplicate code. (Considering the size of the `vf-cxx-ast-exporter` binary, I assume some parts of LLVM have been linked into it statically.)

**Comment by btj** [comment]
The Ubuntu release indeed also fails:
```console
bartj@bartj-VirtualBox:~/verifast-21.04-20-g797e8fcf$ bin/verifast -c tests/cxx/java_like/Stack.cpp
tests/cxx/java_like/Stack.cpp
Fatal error: exception (Failure
   "Cxx AST exporter error:\
  \n/home/bartj/verifast-21.04-20-g797e8fcf/bin/vf-cxx-ast-exporter: error while loading shared libraries: libLLVM-13.so.1: cannot open shared object file: No such file or directory\
  \n")
```

**Comment by btj** [comment]
> However, for Ubuntu, no release is available for LLVM 13.0 on 18.04 (there is one for 16.04 and 20.04).

I would expect that the release for 16.04 will also work on 18.04?

**Comment by NielsMommen** [comment]
> Another option would be to simply ship the LLVM DLL along with VeriFast, like we do with GTK and Z3? The launch script sets DYLD_LIBRARY_PATH to point to VeriFast's lib directory.

> Of course, ideally we avoid shipping duplicate code. (Considering the size of the vf-cxx-ast-exporter binary, I assume some parts of LLVM have been linked into it statically.)

If libLLVM.dylib is used, nothing from LLVM should have been linked statically. The size of libLLVM.dylib is ~100MB and contains a lot of components that we don't use.

When building LLVM/Clang from source on my machine (with dynamic linking turned off), vf-cxx-ast-exporter is ~30MB.

> Another option is to use the LLVM/Clang release from GitHub instead of Homebrew. These releases seem to be built with both  LLVM_BUILD_LLVM_DYLIB and LLVM_LINK_LLVM_DYLIB turned off. The big advantage is that CMake will resolve all cyclic and external dependencies. However, for Ubuntu, no release is available for LLVM 13.0 on 18.04 (there is one for 16.04 and 20.04).

When using the GitHub release on MacOS, the following error is produced:
```
make[3]: *** No rule to make target '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/lib/libcurses.tbd', needed by 'vf-cxx-ast-exporter'.  Stop.
```
whilst the cmake setup detects libcurses at the correct location:
```
-- Found Curses: /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/usr/lib/libcurses.tbd  
```
